apps/src/componentLibrary/dialog/Dialog.tsx

```diff
@@ -26,63 +39,101 @@ export interface DialogProps extends HTMLAttributes<HTMLDivElement> {
   closeLabel?: string;
   /** Dialog icon */
   icon?: FontAwesomeV6IconProps;
+  /** Dialog image url */
+  imageUrl?: string;
 }
 
 /**
  * ## Production-ready Checklist:
- *  * (?) implementation of component approved by design team;
- *  * (?) has storybook, covered with stories and documentation;
- *  * (?) has tests: test every prop, every state and every interaction that's js related;
+ *  * (✔) implementation of component approved by design team;
+ *  * (✔) has storybook, covered with stories and documentation;
+ *  * (✔) has tests: test every prop, every state and every interaction that's js related;
  *  * (see apps/test/unit/componentLibrary/DialogTest.tsx)
  *  * (?) passes accessibility checks;
  *
- * ###  Status: ```WIP```
+ * ###  Status: ```Ready for dev```
  *
  * Design System: Dialog Component.
- * Renders Alert to notify user about something.
+ * Renders Dialog window that user should interact with.
  */
 const Dialog: React.FunctionComponent<DialogProps> = ({
   title,
-  type = 'noIcon',
-  content,
-  showSecondaryButton,
+  description,
+  primaryButtonProps,
+  secondaryButtonProps,
+  mode = 'light',
   className,
+  customContent,
+  customBottomContent,
   onClose,
   closeLabel = 'Close dialog',
   icon,
+  imageUrl,
   ...HTMLAttributes
 }) => {
+  const dialogRef = useRef<HTMLDivElement>(null);
+
+  useBodyScrollLock(true);
+  useFocusTrap(dialogRef);
+  useEscapeKeyHandler(onClose);
```

Comment:
To confirm, these would no longer be needed if/when we can use the `dialog` element here right?

Reply:
```
  useBodyScrollLock(true);
  useFocusTrap(dialogRef);
```
  Should not be needed for `dialog` element
  
`  useEscapeKeyHandler(onClose);` will still be needed